charts/misc-apps/templates/downscaler.yaml

```diff
@@ -0,0 +1,33 @@
+{{ if .Values.downscaler.enabled }}
+{{ template "argoconfig.application" (list . "misc-apps.downscaler") }}
+{{ end }}
+
+{{- define "misc-apps.downscaler" -}}{{- $app := unset .Values.downscaler "enabled" -}}{{- $name := default $app.namespace $app.name -}}
+metadata:
+  name: {{ template "common.fullname" . }}-{{ $name }}
```

Comment:
We use nameOverride on all downstream charts to make their names short for exactly this reason.

It uses this:

https://github.com/adfinis-sygroup/helm-charts/blob/6de403d5c319e0849a8a11b9d9cc222c75fc9e6d/charts/common/templates/_fullname.tpl#L22

Does that add up with your local results?

We might want to replace the `"common.fullname"` template with a yet to be written `"argoconfig.appname"` template... I'll have to think about this a bit until I come up with something I'm happy about.

Reply:
As it turns out a fix for this was already lurking in argoconfig... 

`helm template -f examples/kube-downscaler.yaml .` now generates `release-name-misc-apps-downscaler` which has to be fine for now.

Reply:
Yes, I can agree that this is fine for now, also since I cannot provide a better solution. (Also, the "infra" part that might be rather misleading for some misc apps is gone now.)
After looking at your comment in a bit more detail I will also suggest providing a `fullnameOverride` in the values when deploying this for our test cluster. (This allows to remove the redundancy and creating more sensible names for the final applications in Argo CD.)